tools/package-docs-content/index.bzl

```diff
@@ -29,25 +30,27 @@ def _package_docs_content(ctx):
 
         for input_file in section_files:
             # Creates a relative path from the input file. We don't want to include the full
-            # path in docs content. e.g. `docs-content/overviews/cdk/src/cdk/a11y/a11y.html`.
-            # Instead, we want the path to be: `docs-content/overviews/cdk/a11y/a11y.html`.
-            section_relative_file_name = input_file.short_path[len(base_dir):]
+            # path in docs content. e.g. `/docs-content/overviews/cdk/src/cdk/a11y/a11y.html`. Instead,
+            # we want the path to be: `/docs-content/overviews/cdk/a11y/a11y.html`.
+            section_relative_file_name = input_file.short_path[len(base_dir) + 1:]
```

Comment:
Hmm, can you explain what this change was for? The comment isn't changed so I'm curious what this was about

Reply:
Previously, we declared outputs on a per-file basis in Bazel as we knew what outputs to expect from a set of input files. This does no longer work, now that input file could also be a Bazel tree artifact. For those we cannot determine outputs at Bazel analysis time.

We fix this (in a Bazel idiomatic way), by not declaring outputs per-file, but rather by creating a single tree artifact output (see `declare_directory`) that can hold arbitrary files. Due to this, we can remove the `declare_file` code here.